docs/DeveloperGuide.md

```diff
@@ -68,33 +77,29 @@ list is simply represented as a `TaskList` instead of a full-fledged `Module`.
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/CommandClassDiagram.puml)
 
-Here is a simplified class diagram illustrating two example commands:  
-![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/ngys117/tp/branch-PR-DeveloperGuide/docs/CommandClassDiagram.puml)
+All commands inherit the abstract `Command` class and must contain an `execute()` method. The program logic that must be executed to fulfil the requested command is implemented in this method. Additionally, `execute()` returns any command output to be displayed to the user as feedback.
 
-How the command executes:  
-The type of action that a command executes is dependent on which constructor is called and values passed to it by the respective parser.
+`Command` instances are generated by their corresponding `Parser` classes (e.g. `AddCommand` is constructed by `AddParser`) and executed by `Main`.
 
 ### Storage Component
-**API**: [Storage.java](https://github.com/AY2122S2-CS2113T-T10-3/tp/tree/master/src/main/java/seedu/duke/storage/Storage.java) <br>
 
-The storage interface is implemented by JsonStorage in Mod Happy, which reads and loads data in json format.  
-Here is a class diagram on `Storage`:
+The `Storage` component is responsible for the saving and loading of program data from and to its data files. The following class diagram illustrates the structure of this component:
 
 ![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/Storage.puml)
 
-How data is saved and loaded:  
-ListStorage saves an ArrayList of any class that extends Object in json format, and loads them back into corresponding objects. (e.g. ModuleListStorage, TaskListStorage inherit from ListStorage).  
-There is navigability to Storage from Main and SaveCommand, which handles the load and write data to/from disk respectively.
+Several type-specific classes exist, each overseeing the storage of a different type of user data: 
+
+* `ConfigurationStorage` handles the saving and loading of user preferences. This data is stored in the `data/configuration.json` file.
+* `TaskListStorage` handles the saving and loading of the General Tasks list as an `ArrayList<Task>` instance. This data is stored in the `data/tasks.json` file.
+* `ModuleListStorage` handles the saving and loading of all user-created modules as well as the tasks associated with them as an `ArrayList<Module>` instance. This data is stored in the `data/modules.json` file.
+
+All write operations rely on the general purpose `writeData()` method of the abstract class `JsonStorage`. However, read operations are implemented in each type-specific class; the `readData()` methods of these classes reconstruct the original object from the serialised data and return them.
```

Comment:
I remember you commented that the architecture part should keep in a bit more high level? Then why modified this back?

Reply:
After the edits were removed, I realised that there wasn't enough to talk about for the component at all...

After looking at past semesters' DGs, I realised that some of them do indeed explain what the subclasses in each package are for and what they do. So I decided to add the content from your original PR back, otherwise there is nothing much to say for this section.

Reply:
But our previous plan is to put these information in the implementation section, right?

Reply:
We can probably move this + the sequence diagram to the implementation section under a new section like "parser implementation" later on. I will put it on the google doc so we remember